resources/component.rb

```diff
@@ -0,0 +1,75 @@
+property :component_name, ::String, name_property: true # maven2.artifactId / raw.asset1.filename
```

Comment:
OPT: instead of having "specific params" you could have created a generic component resource + specific ones that extend it 
IMHO it would have been cleaner that all the "case"

Reply:
True, can be quickly extended later